### ISSUE TITLE
systemd-homed pulls in /home even if noauto is specified in fstab

### ISSUE THREAD

**Issue by windmueller** [systemd/systemd#15102]
**systemd version the issue has been seen with**
245

**Used distribution**
 Arch Linux

**Expected behaviour you didn't see**
systemd-fsck should ignore fstab entries with passno == 0

**Unexpected behaviour you saw**
During system boot I have to wait additional 90 seconds until the timeout for systemd-fsck is reached which tries to check my unmounted LUKS partition:

```
systemd[1]: dev-mapper-home\x2dusername.device: Job dev-mapper-home\x2dusername.device/start timed out.
systemd[1]: Timed out waiting for device /dev/mapper/home-username.
systemd[1]: Dependency failed for /home.
systemd[1]: Dependency failed for Home Manager.
systemd[1]: systemd-homed.service: Job systemd-homed.service/start failed with result 'dependency'.
systemd[1]: home.mount: Job home.mount/start failed with result 'dependency'.
systemd[1]: dev-mapper-home\x2dusername.device: Job dev-mapper-home\x2dusername.device/start failed with result 'timeout'.
```
The corresponding entry in fstab has passno == 0, but the service is started on every boot.

```
# grep mapper /etc/fstab | awk '{ print $4, $5, $6 }'
rw,noatime,noauto 0 0

# systemctl status --full systemd-fsck@dev-mapper-home\x2dusername.service
     systemd-fsck@dev-mapper-homex2dusername.service - File System Check on /dev/mapper/homex2dusername
     Loaded: loaded (/usr/lib/systemd/system/systemd-fsck@.service; static; vendor preset: disabled)
     Active: inactive (dead)
     Docs: man:systemd-fsck@.service(8)
```

**Steps to reproduce the problem**

* Create a LUKS partition
* Add the mapper device to `/etc/fstab`
* Reboot the system

**Comment by arvidjaar** [comment]
> timeout for systemd-fsck is reached

There is no `fsck` anywhere in logs you posted. What makes you believe it has anything to do with `fsck` in the first place?

Did you configure autostart of LUKS device in `/etc/crypttab`?

**Comment by stovocor** [comment]
> There is no `fsck` anywhere in logs you posted. What makes you believe it has anything to do with `fsck` in the first place?

You are correct, it seems that I made an error interpreting the log output.

However, I still have no clue why the boot time shows this sudden increase.

> Did you configure autostart of LUKS device in `/etc/crypttab`?

No, the file only contains comments. The actual mount is done via PAM.

**Comment by arvidjaar** [comment]
> I still have no clue why the boot time shows this sudden increase

According to logs you provided, device where /home is located is not available during boot.

**Comment by stovocor** [comment]
> According to logs you provided, device where /home is located is not available during boot.

That is correct, the mapper device will not be created until I login.

It seems that one of the latest updates changed the behavior. How do I tell systemd to ignore this partition?

**Comment by arvidjaar** [comment]
> How do I tell systemd to ignore this partition?

Do not add it to `/etc/fstab`. `sytemd-homed` has dependency on `home` mount point.

**Comment by stovocor** [comment]
> Do not add it to `/etc/fstab`.

Okay, removing the line fixed my issue somehow. I still have to mount the partition now in the PAM script.

> `sytemd-homed` has dependency on `home` mount point.

Has this changed lately? Before one of the latest updates my system booted without delay.

Furthermore, should systemd honor the "noauto" option in my fstab?

**Comment by arvidjaar** [comment]
> Has this changed lately?

Yes, `systemd-homed` is new in v245.

> should systemd honor the "noauto" option in my fstab?

It does. `noauto` just means "do not add mount unit as dependency of `local-fs.target`", nothing more. `systemd` still creates mount unit for every line in `/etc/fstab` and `systemd-homed` has `RequiresMountsFor=/home` which pulls in mount unit generated from `/etc/fstab`.

You would have the same issue with native mount unit for home. It is entirely unrelated to `noauto` flag or the fact that it is in `/etc/fstab` at all.

**Comment by stovocor** [comment]
It seems to be unclear if this is a bug or not.

Let me try to summarize: Is there any way to prevent the timeout if the `/home` partition in `/etc/fstab` is only available after an interactive login?

**Comment by poettering** [comment]
even though i doubt this kind of setup is really something we should support (i.e. systemd-user-session.service at the latest should mean /home is available), I guess we can soften the dep from homed a bit, and make it a simple After= instead of also having Requires= on home.mount.

**Comment by poettering** [comment]
Fix waiting in #15277.
